File: cookbook/datetime/go/formatting.go
package datetime

import "time"

func ToISO8601(t time.Time) string {
	return t.Format(time.RFC3339)
}

func ToISO8601Nano(t time.Time) string {
	return t.Format(time.RFC3339Nano)
}

func ToDateTime(t time.Time) string {
	return t.Format(time.DateTime)
}

func ToDateOnly(t time.Time) string {
	return t.Format(time.DateOnly)
}

File: cookbook/datetime/typescript/formatting.ts
export function toISO8601(date: Date): string {
  return date.toISOString();
}

export function toDateTimeString(date: Date, timeZone: string = "UTC"): string {
  const options: Intl.DateTimeFormatOptions = {
    year: "numeric",
    month: "2-digit",
    day: "2-digit",
    hour: "2-digit",
    minute: "2-digit",
    second: "2-digit",
    hour12: false,
    timeZone: timeZone,
  };
  let formatter: Intl.DateTimeFormat;
  try {
    formatter = new Intl.DateTimeFormat("sv-SE", options);
  } catch (e) {
    if (e instanceof RangeError) {
      throw new RangeError(
        `toDateTimeString() argument timeZone must be valid IANA time zone string, got: ${timeZone}`
      );
    }
    throw e;
  }
  return formatter.format(date);
}
